File: Java/Reverse Integer.java
class Solution {
    public int reverse(int x) 
    {
     int f = 1;
        if(x<0)
        {
            f = -1;
            x *= -1;
        }
        StringBuilder s = new StringBuilder(String.valueOf(x));
        try
        {
        s = s.reverse();
        int ans = Integer.valueOf(String.valueOf(s)) *f;
        if(Math.pow(-2,31)>ans || Math.pow(2,31)-1<ans)return 0;
        return ans;
        }catch(Exception e)
        {
            return 0;
        }
    }
}

File: C/Reverse Integer.c


int reverse(int x){
    
long l,s,r,f;
    
    f=pow(2,31)-1;
    if(x<0)
        l=(long)x * (-1);
    else
        l=(long)x;
    s=0;
    while(l!=0)
    {
        r=l%10;
        s=s*10+r;
        l=l/10;
        if(f<s)
            return 0;
    
    }
    if(x<0)
        s= s * (-1);
    return s;
}